Fixes issue #137

Roshan-Thomas opened the pull request:
**Context:**
Unitary Hack task for updating the tutorial and documentation style.

**Description of the change:**
The following changes are implemented in this PR

- Upgraded to `sphinxawesome_theme` and made sure the changes were reflected in all the pages. 
- The white and dark logos were added for both the dark and light versions of the website.
- The `tutorial` was updated, and both required examples were added to it and deleted from the examples menu. 
- Updated the MBQC and LC-MBQC intros to reflect the requested paper in the issue. 

**Related issue:**
The related issue is #137.

Roshan-Thomas commented:
Hey @shinich1, my apologies for getting back to you pretty late, I had a lot of uni work to do. 

I have completed the required changes,
- updated the code on the introduction and tutorial pages to reflect the latest changes to the codebase.
- made the tutorial page a lot easier to follow, and
- added graph visualizations on the tutorial page. 

The build can be found [here](https://app.readthedocs.org/projects/graphix-uh/builds/24601310/). I ran the build a while back and it still hasn't completed. It successfully installed graphix and then remain stuck there. 

Let me know if there are any more changes needed to be made.

shinich1 commented:
> Hey @shinich1, my apologies for getting back to you pretty late, I had a lot of uni work to do.
> 
> I have completed the required changes,
> 
> * updated the code on the introduction and tutorial pages to reflect the latest changes to the codebase.
> * made the tutorial page a lot easier to follow, and
> * added graph visualizations on the tutorial page.
> 
> The build can be found [here](https://app.readthedocs.org/projects/graphix-uh/builds/24601310/). I ran the build a while back and it still hasn't completed. It successfully installed graphix and then remain stuck there.
> 
> Let me know if there are any more changes needed to be made.

@Roshan-Thomas  Thanks! the docs seems to have built fine (it takes time building all the `examples` from the code). I have a few remaining requests before merging:
- can you make the logo on the top-left corner of the page larger, so that it's similar size as [current one ](https://graphix.readthedocs.io/en/latest/index.html). please remove the bold text next to the logo reading 'graphix documentation'.
- can you set up code highlighting across the docs? all code blocks are black, which is a little difficult to read.
- the heading level for 'simulating noisy mbqc', running pattern on quantum devices and generating qasm file need to be one level higher - see screenshot below, it currently looks as if they are part of 'optimizing patterns'
<img width="316" alt="image" src="https://github.com/TeamGraphix/graphix/assets/33350509/499198ea-5f09-4062-936f-30f70714ebe7">

shinich1 commented on docs/source/tutorial.rst:
```diff
@@ -141,7 +172,12 @@ X byproduct, node = 7, domain = [6]
+    :alt: visualization of optimized pattern
+
 
+As we see below, we can significantly optimize this for better simulation performance and even operations in quantum hardware.
```
can you change this line to 
`As we see below, we can simplify and optimize the pattern by calling various methods of `~graphix.pattern.Pattern` class.

Roshan-Thomas replied:
added to PR

shinich1 commented on docs/source/tutorial.rst:
```diff
@@ -155,31 +191,33 @@ These can be called with :meth:`~graphix.pattern.Pattern.standardize` and :meth:
+    :scale: 60 %
+    :alt: pattern visualization after standarziation
 
 The command sequence is now much clear and note that the byproduct commands now only apply to output nodes (3, 7).
```
change this line to 
`The command sequence is now simpler and note that the most byproduct commands now apply to output nodes (3, 7).`

Roshan-Thomas replied:
added to PR

Roshan-Thomas commented:
Hi @shinich1, I have added the following changes,

- removed 'graphix documentation' on top-left corner
- added code highlighting
- changed tutorial heading levels as per request
- updated the tutorial page to have simpler language as per request.

Note: The logo was something I was not able to enlarge it as the `spinxawesome-theme` does not have that feature. I have searched in the documentation and there is no mention on how to increase the size of the logo. If you know of the method to do it, please let me know and I can implement it.

shinich1 commented:
> Hi @shinich1, I have added the following changes,
> 
> * removed 'graphix documentation' on top-left corner
> * added code highlighting
> * changed tutorial heading levels as per request
> * updated the tutorial page to have simpler language as per request.
> 
> Note: The logo was something I was not able to enlarge it as the `spinxawesome-theme` does not have that feature. I have searched in the documentation and there is no mention on how to increase the size of the logo. If you know of the method to do it, please let me know and I can implement it.

@Roshan-Thomas Thanks a lot for looking into this. just a final request, would `furo` theme allow you to enlarge the logo? could you try with [this](https://github.com/TeamGraphix/graphix/blob/master/docs/logo/black_with_name.png) logo file?

shinich1 commented:
e.g. `quimb` docs has nicely-sized logo and they seem to use `furo` theme.

Roshan-Thomas commented:
@shinich1 I have updated the theme to `furo` and got the logo (dark and light themed) to work. 

Also, I had to place the `dark` and `light` logo in the `_static` folder as `furo` reads the images from the `_static` folder only, and for some odd reason the file paths to the `logo` folder was not working. 

Hope this fixes all the issues and the changes required.

shinich1 commented:
> @shinich1 I have updated the theme to `furo` and got the logo (dark and light themed) to work.
> 
> Also, I had to place the `dark` and `light` logo in the `_static` folder as `furo` reads the images from the `_static` folder only, and for some odd reason the file paths to the `logo` folder was not working.
> 
> Hope this fixes all the issues and the changes required.

nice, it looks like the build failed - could you fix it?
https://app.readthedocs.org/projects/graphix-uh/

shinich1 commented:
> fixed the issue, the `furo` package was not present in the `requirements.txt`. I have added it now (see last commit).

will check once the build completed. I see lint CI failing below, could you `black -l 120` the `conf.py`?

Roshan-Thomas commented:
> > fixed the issue, the `furo` package was not present in the `requirements.txt`. I have added it now (see last commit).
> 
> will check once the build completed. I see lint CI failing below, could you `black -l 120` the `conf.py` (see [our guide](https://github.com/TeamGraphix/graphix/blob/master/CONTRIBUTING.md))?

should I run the lint for any other files (other than `conf.py`)?

shinich1 commented:
@Roshan-Thomas all look good now - I will merge this once tests passed and I checked things again. 
btw, I'd appreciate if you could star the repo, if not done yet :)

Roshan-Thomas commented:
> @Roshan-Thomas all look good now - I will merge this once tests passed and I checked things again. btw, I'd appreciate if you could star the repo, if not done yet :)

Thank you!
